Library/Homebrew/livecheck/strategy/sparkle.rb

```diff
@@ -86,6 +86,16 @@ def self.item_from_content(content)
 
             next if (os = enclosure&.attr("os")) && os != "osx"
 
+            if OS.mac? && (minimum_system_version = (item > "minimumSystemVersion").first&.text&.strip)
```

Comment:
```suggestion
            if (minimum_system_version = (item > "minimumSystemVersion").first&.text&.gsub(/[^0-9.]/, ""))
```

We can drop `OS.mac?` since this code isn't platform-dependent. I've also added some code that removes anything that isn't a number or period. This takes care of the previous `#strip` and also removes any preceding `v` without being explicit about it.

Reply:
what about underscores?

Reply:
Of the casks currently using `Sparkle`, all of the encountered `minimumSystemVersion` values use a format consisting only of numbers and periods (e.g., `10.14`, `10.14.0`, `10.14.4`, etc.), with the sole exception of the three beginning with `v` (`v10.11`, `v10.13`, and `v10.14`).

A version string containing an underscore wouldn't be valid input for `OS::Mac::Version#new` and would give a `MacOSVersionError`, as the input string needs to match the strict regex it uses for validation (`/\A1\d+(?:\.\d+){0,2}\Z/`).

Reply:
> A version string containing an underscore wouldn't be valid input for `OS::Mac::Version#new` and would give a `MacOSVersionError`, as the input string needs to match the strict regex it uses for validation (`/\A1\d+(?:\.\d+){0,2}\Z/`).

Thanks, @samford . I guess I was thinking of something else here. Sorry for the confusion!